File: src/main/java/org/example/expert/domain/common/service/HealthCheckService.java
package org.example.expert.domain.common.service;

import jakarta.persistence.EntityManager;
import jakarta.persistence.PersistenceContext;
import org.springframework.stereotype.Service;

@Service
public class HealthCheckService {
    @PersistenceContext
    private EntityManager entityManager;

    public boolean isConnected() {
        try {
            return entityManager.createNativeQuery("SELECT 1").getSingleResult() != null;
        } catch (Exception e) {
            return false;
        }
    }
}

File: src/main/java/org/example/expert/domain/common/service/HealthCheckService.kt
package org.example.expert.domain.common.service

import jakarta.persistence.EntityManager
import jakarta.persistence.PersistenceContext
import org.springframework.stereotype.Service

@Service
class HealthCheckService(
    @PersistenceContext
    private val entityManager: EntityManager
) {
    val isConnected: Boolean
        get() {
            return try {
                entityManager.createNativeQuery("SELECT 1").singleResult != null
            } catch (e: Exception) {
                false
            }
        }
}